### PR TITLE
Rockset Query Runner: Fetch collections when api key is access limited

### PR DESCRIPTION
**Pull request by MaxBer**
When the rockset api key does not have full access to all workspaces the collection fetching fails. This PR checks for a result when fetching collection columns.

## What type of PR is this? 

- [X] Bug Fix

## Description
When the rockset api key does not have full access to all workspaces the collection fetching fails. This PR checks for a result when fetching collection columns.

Expected compatibility: Rockset production db.

## How is this tested?

- [X] Manually

1. Build image
2. Set up connection
3. Test connection
4. Go to schema overview in /queries/new

Refresh works although our API key does not have read access to all collections (across workspaces)

### CODE REVIEW

**Review by masayuki038 — CHANGES_REQUESTED** [review]
Sorry, I didn't submit my review. I left a comment. Could you check it?

File: redash/query_runner/rockset.py [hunk 1]
```diff
@@ -100,10 +101,12 @@ def _get_tables(self, schema):
         for workspace in self.api.list_workspaces():
             for collection in self.api.list_collections(workspace):
                 table_name = collection if workspace == "commons" else "{}.{}".format(workspace, collection)
-                schema[table_name] = {
-                    "name": table_name,
-                    "columns": self.api.collection_columns(workspace, collection),
-                }
+                columns = self.api.collection_columns(workspace, collection)
+                if columns:
```
[the remark this answers is not in the document]
Comment (MaxBer) [reply redash/query_runner/rockset.py:105]: I personally feel it doesn't make sense to show collections & tables where the user does not have access to the data?

[the remark this answers is not in the document]
Comment (MaxBer) [reply redash/query_runner/rockset.py:105]: Calling list_workspaces & list_collections requires an access level within Rockset which yields all the workspaces available (even ones I do not have "data read" access to), and the request does not seem to provide me with any indication if I have access or not.

In this case the role has access to read data in a specific workspace but not the other ones, and when it attempts to list a collection where I do not get a result.

**Review by masayuki038 — APPROVED** [review]
@MaxBer Thanks for your explanation. I've understood the situation. The Rockset API specification looks a little strange, but it's not something we can change, so that's okay from my end.

I checked the test coverage in the previous commit since some uncovered code existed. Most of it wasn't covered from last time. So, I approve this MR.
https://app.codecov.io/gh/getredash/redash/commit/a8c6dd0043c06b3e90533bcc9eed8b5a4c9bf65c/blob/redash/query_runner/rockset.py

Please wait to finish the reviews for a few days.

@guidopetri @konnectr Could you check this if you have some time?

**Review by guidopetri — APPROVED** [review]
This looks reasonable to me too, though I'd prefer a test being added. @MaxBer any interest in adding a test?

**Comment by MaxBer** [comment]
@guidopetri Sorry just saw this. We ended up going with BigQuery and shut down our Rockset instances, so.I don't really have the means to develop this further unfortunately.

**Comment by justinclift** [comment]
> I don't really have the means to develop this further unfortunately.

No worries.  Lets close this PR for now.

Maybe future people using Rockset will be able to build on it anyway. :smile:

**Pull request closed without merging**